js/ui/search.js

```diff
@@ -174,7 +174,7 @@ const ListSearchResult = new Lang.Class({
                                y_align: St.Align.MIDDLE });
 
         let title = new St.Label({ style_class: 'list-search-result-title',
-                                   text: this.metaInfo['name'],
+                                   text: this.metaInfo['name'] || '',
```

Comment:
Yeah, I debated merging this too... The other option is to just filter out meta infos with no name. The shell definitely should not throw exceptions or crash when the provider sends malformed results.
@magcius Would you prefer the other approach?